transmart-core-db/src/integration-test/groovy/org/transmartproject/db/multidimquery/QueryServicePgSpec.groovy

```diff
@@ -619,4 +607,77 @@ class QueryServicePgSpec extends Specification {
         }
     }
 
+    void 'test aggregates'() {
+        def expectedMax = 102
+        def expectedMin = 56
+        def expectedAverage = 74.78
+        def expectedCount = 9
+        def expectedPatientCount = 3
+        def expectedStandardDiviation = 14.7
+        def user = User.findByUsername('test-public-user-1')
+        def heartRate = new ConceptConstraint(
+                path: '\\Public Studies\\EHR\\Vital Signs\\Heart Rate\\')
+
+        when:
+        def maxMap = multiDimService.aggregate(EnumSet.of(MAX), heartRate, user)
+        then:
+        maxMap == [ (MAX): expectedMax ]
+
+        when:
+        def minMap = multiDimService.aggregate(EnumSet.of(MIN), heartRate, user)
+        then:
+        minMap == [ (MIN): expectedMin ]
+
+        when:
+        def avgMap = multiDimService.aggregate(EnumSet.of(AVERAGE), heartRate, user)
+        then:
+        avgMap.size() == 1
+        AVERAGE in avgMap
+        avgMap[AVERAGE].round(2) == expectedAverage
+
+        when:
+        def countMap = multiDimService.aggregate(EnumSet.of(COUNT), heartRate, user)
+        then:
+        countMap == [ (COUNT): expectedCount ]
+
+        when:
+        def patientCountMap = multiDimService.aggregate(EnumSet.of(PATIENT_COUNT), heartRate, user)
+        then:
+        patientCountMap == [ (PATIENT_COUNT): expectedPatientCount ]
+
+        when:
+        def sdMap = multiDimService.aggregate(EnumSet.of(STD_DEV), heartRate, user)
+        then:
+        sdMap.size() == 1
+        STD_DEV in sdMap
+        sdMap[STD_DEV].round(2) == expectedStandardDiviation
+
+        when:
+        def compositeMap = multiDimService.aggregate(EnumSet.allOf(AggregateFunction), heartRate, user)
+        then:
+        compositeMap.size() == 6
+        compositeMap[MAX] == expectedMax
+        compositeMap[MIN] == expectedMin
+        compositeMap[AVERAGE].round(2) == expectedAverage
+        compositeMap[COUNT] == expectedCount
+        compositeMap[PATIENT_COUNT] == expectedPatientCount
+        compositeMap[STD_DEV].round(2) == expectedStandardDiviation
+    }
+
+    void 'test categorical value frequencies'() {
+        def user = User.findByUsername('test-public-user-1')
+        def race = new ConceptConstraint(
+                path: '\\Public Studies\\CATEGORICAL_VALUES\\Demography\\Race\\')
+        when:
+        def categoricalValueFreqMap = multiDimService.countCategoricalValues(race, user)
+        then:
+        categoricalValueFreqMap == [ Caucasian: 2L, Latino: 1L ]
+
+        when: 'only numerical observations selected'
+        def emptyMap = multiDimService.countCategoricalValues(new ConceptConstraint(
+                path: '\\Public Studies\\EHR\\Vital Signs\\Heart Rate\\'), user)
+        then: 'map is empty'
```

Comment:
I am not sure if I expect an error or an empty map here.

Reply:
What happens when doing a numerical aggregate on a categorical variable? Also empty map, right?

Reply:
yes